### PR TITLE
Add disabledThumbColor and other properties from Switch to SwitchListTile

### PR DESCRIPTION
**Pull request by escamoteur**
SwitchListTile did not expose all properties that a normal Switch does as reported here

#18753

So I added them

### CODE REVIEW

File: packages/flutter/lib/src/material/switch_list_tile.dart [hunk 1]
```diff
@@ -82,10 +85,10 @@ class SwitchListTile extends StatelessWidget {
...
     this.selected = false,
   }) : assert(value != null),
-       assert(isThreeLine != null),
-       assert(!isThreeLine || subtitle != null),
-       assert(selected != null),
-       super(key: key);
+        assert(isThreeLine != null),
+        assert(!isThreeLine || subtitle != null),
+        assert(selected != null),
+        super(key: key);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch_list_tile.dart]: please leave the indentation as before, thanks

File: packages/flutter/lib/src/material/switch_list_tile.dart [hunk 2]
```diff
@@ -171,6 +190,10 @@ class SwitchListTile extends StatelessWidget {
       activeColor: activeColor,
       activeThumbImage: activeThumbImage,
       inactiveThumbImage: inactiveThumbImage,
+      activeTrackColor: activeTrackColor,
+      inactiveTrackColor: inactiveTrackColor,
+      inactiveThumbColor: inactiveThumbColor,
+
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch_list_tile.dart]: nit: extraneous new line

**Comment by Hixie** [comment]
This will need tests.

**Comment by escamoteur** [comment]
Corrected the formatting. Could you guide me on how to add tests?

**Comment by Hixie** [comment]
@gspencergoog can you help @escamoteur out here? Thanks!

**Comment by gspencergoog** [comment]
Sure thing.

To create the test, you'll need to create a new file: `packages/flutter/test/material/switch_list_tile_test.dart`.  You'll start with the following boilerplate for creating a test:

```dart
// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  testWidgets('SwitchListTile has the right colors', (WidgetTester tester) async {
    await tester.pumpWidget(
      // you might need to put this into a ListView, of course.
      new SwitchListTile(/*...*/),
    );
    expect(/*...*/);
  });
}
```

Where you `expect` that the colors of the track and thumb are what you expect them to be.

You can do this by finding the child Switch for the `SwitchListTile` using something similar to `final Switch switch = tester.widget(find.byType(Switch));`, and then expecting that the colors are what you expect them to be.

That should get you started.  Let me know if you need more help.

**Comment by gspencergoog** [comment]
So, are you saying that the code doesn't work ("it does not update the colors"), or that the test doesn't do what you expected? The tests currently aren't passing: is that what you meant?

**Comment by escamoteur** [comment]
Sorry for being unclear. The test doesn't do what I expected. I get the opposite colors that I expect for enabled/disabled and have no idea why.

**Comment by Hixie** [comment]
@escamoteur Any luck debugging the tests? What would you like the next step here to be? If you want we can close this PR and we will eventually one day get back to it, but it might be a while.

**Comment by escamoteur** [comment]
@Hixie Honestly I was hoping someone fro your side could have a look at this because I could not find a solution.

**Review by mockturtl — COMMENTED** [review]
The colors look good to me.

Initially, in the "off" state, I see a blue track with a yellow thumb.

After the tap, in the "on" state, I see a green track with a red thumb.

**Comment by Hixie** [comment]
cc @jslavitz this might be a good one to run with next if you're looking for another needs love PR.

**Comment by jslavitz** [comment]
FIxed this in the above PR so I'm closing this!

**Comment by jslavitz** [comment]
Hey sorry about that! So it's not possible in Github to edit/append to PRs that aren't authored by you, so I had to make a new PR. I've added your name in the comment of the PR that I wrote so that people will know it was originally made by you. Also, all of the commit history carries over, so if you want to use the PR I published as proof of your work people will be able to see exactly what code you wrote vs. what I did. Again, we really appreciate your contribution!

**Comment by escamoteur** [comment]
Thanks

**Pull request closed without merging**